docs: blitz (#3479)
ibis-project/ibis: docs/backends/index.md

# Backends

## String Generating Backends

> The first category of backend translate Ibis expressions into strings.

Reviewer: ```suggestion
The first category of backend translate Ibis expressions into string queries.
```

> Generally speaking these backends handle their own execution.

Reviewer: That's not necessarily true since pyspark and datafusion handles their own execution as well but we utility a different API to talk with them. 

I'd say that these backends are databases with a SQL interface.

Author: I think this sentence that I've written here is ambiguous and possibly just wrong, I'll reword it.

Author: I'm dubious on the value of this entire file, as it's a wall of text that no one is likely to get much value out of.

Reviewer: It would be useful for ibis contributors though.

Author: Yeah, we can keep it for now.

> The compiler turns each expression into a string and passes that string to the

Reviewer: ```suggestion
The compiler turns each expression into a string query and passes that query to the
```

database through a driver API.

- [Apache Impala](Impala.md)
- [ClickHouse](ClickHouse.md)
- [Google BigQuery](https://github.com/ibis-project/ibis-bigquery/)
- [OmniSciDB](https://github.com/omnisci/ibis-omniscidb)

## Expression Generating Backends

> The next category of backends translates ibis expressions into other

Reviewer: ```suggestion
The next category of backends translates ibis expressions into other expression based representations. 
```

Author: 🤷🏻 I could go either way on this. I'm not sure what additional information is conveyed by `expression based representations` versus `expressions`.

expressions.

Instead of generating strings for each expression these backends produce
another kind of expression and typically have high-level APIs for execution.

- [Dask](Dask.md)
- [Datafusion](Datafusion.md)

> - [MySQL](MySQL.md)

Reviewer: I'd move the SQLAlchemy based backends to the previous category since we eventually compile the SQLA queries to strings queries as well, just use SQLA as an intermediate representation due to its broad use.

Author: We don't though, the fact that SQLAlchemy does is an implementation detail that ibis is pretty much ignorant of.

- [PostgreSQL](PostgreSQL.md)
- [PySpark](PySpark.md)
- [SQLite](SQLite.md)

## Direct Execution Backends

> The pandas backend is the only direction execution backend. A full description

Reviewer: "Direction execution"?

Author: Should be "direct" 😂

Author: Done.
